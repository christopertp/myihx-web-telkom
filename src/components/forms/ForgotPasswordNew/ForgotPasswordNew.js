import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Field } from 'redux-form';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import PasswordField from '../../fields/Password';
import styles from './styles.scoped.css';

export default function ForgotPasswordNew(props) {
  const { handleSubmit, invalid } = props;
  const { isLoading } = useSelector(s => s.forgotPassword);
  const inputProps = {
    label: 'NEW PASSWORD',
    placeholder: 'Enter your new password',
  };

  return (
    <form className={styles.root} onSubmit={handleSubmit}>
      <Field component={PasswordField} inputProps={inputProps} name="password" />
      <small className={styles.info}>Min. 6 characters with a mix of letters and numbers</small>
      <Button disabled={invalid || isLoading} type="submit">
        {isLoading ? <Spinner /> : 'Create new password'}
      </Button>
    </form>
  );
}

ForgotPasswordNew.defaultProps = {
  handleSubmit: () => {},
  invalid: false,
};

ForgotPasswordNew.propTypes = {
  handleSubmit: PropTypes.func,
  invalid: PropTypes.bool,
};
