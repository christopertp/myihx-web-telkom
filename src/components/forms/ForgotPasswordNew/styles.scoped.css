.root > button {
  margin-top: 2rem;
  width: 100%;
}

.info {
  color: #66686a;
  display: block;
  font-size: 0.75rem;
  margin-top: 0.5rem;
}
