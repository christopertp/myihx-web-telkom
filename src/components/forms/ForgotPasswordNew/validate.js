import { isPassword } from '../../../utils/validation';

export default function validate(values) {
  const { password } = values;

  return {
    password: !isPassword(password) ? 'Minimal 6 karakter terdiri dari gabungan huruf dan angka' : '',
  };
}
