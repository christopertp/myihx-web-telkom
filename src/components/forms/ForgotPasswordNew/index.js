import { reduxForm } from 'redux-form';
import ForgotPasswordNew from './ForgotPasswordNew';
import validate from './validate';

export default reduxForm({
  initialValues: {
    password: '',
  },
  form: 'forgotPasswordNew',
  validate,
})(ForgotPasswordNew);
