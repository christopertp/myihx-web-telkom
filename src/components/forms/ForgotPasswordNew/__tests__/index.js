import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import ForgotPasswordNew from '../ForgotPasswordNew';

useSelector.mockImplementation(fn => {
  fn({});
  return { isLoading: false };
});

describe('src/components/forms/ForgotPasswordNew', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ForgotPasswordNew />);
    expect(tree).toMatchSnapshot();

    useSelector.mockImplementationOnce(() => ({ isLoading: true }));
    ForgotPasswordNew.defaultProps.handleSubmit();
    const result = ForgotPasswordNew({
      ...ForgotPasswordNew.defaultProps,
    });
    const button = result.props.children[2];
    expect(button.props.children.type).toBe(Spinner);
  });
});
