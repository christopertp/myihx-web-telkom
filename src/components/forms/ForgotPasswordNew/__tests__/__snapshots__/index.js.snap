// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/ForgotPasswordNew render 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component={[Function]}
    inputProps={
      Object {
        "label": "NEW PASSWORD",
        "placeholder": "Enter your new password",
      }
    }
    name="password"
  />
  <small>
    Min. 6 characters with a mix of letters and numbers
  </small>
  <Button
    className=""
    disabled={false}
    onClick={null}
    to=""
    type="submit"
    variant="red"
  >
    Create new password
  </Button>
</form>
`;
